src/confirmation-dialog/useConfirmationDialog.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useCallback, useState } from "react";
+import { ConfirmationDialog, ConfirmationDialogProps } from "./ConfirmationDialog";
+
+export interface ConfirmationDialogHookState extends Omit<ConfirmationDialogProps, "isOpen"> {
+    autoClose?: boolean;
+}
+
+export type ConfirmationDialogHookResult = [
+    (props: ConfirmationDialogProps) => JSX.Element | null,
+    (state: ConfirmationDialogHookState) => void,
+    () => void
+];
+
+export function useConfirmationDialog(
+    initialState: ConfirmationDialogHookState | null = null
+): ConfirmationDialogHookResult {
+    const [modalState, updateModal] = useState<ConfirmationDialogHookState | null>(initialState);
+    const closeModal = useCallback(() => updateModal(null), []);
+
+    const component = useCallback(
```

Comment:
The first doubt that comes to mind: is it ok to return a component from a hook? They typically encapsulate behaviour, not view components. I found some discussions online, i.e: https://www.reddit.com/r/reactjs/comments/9yq1l8/how_do_you_feel_about_a_hook_returning_components/

The first obvious prevention people express is that the component identify may be new on each render. We have `useCallback`, which caches the component, but when `modalState` the component is created anew, right?

Not sure if possible, but ideally we'd have a normally defined component on the root level of the module that the hook uses. This component is very lean, so it's not really important here, just as a general concept.